bundles/automation/org.eclipse.smarthome.automation.json.parsers/src/main/java/org/eclipse/smarthome/automation/parser/json/RuleJSONParser.java

```diff
@@ -0,0 +1,305 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ProSyst Software GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.smarthome.automation.parser.json;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.automation.Action;
+import org.eclipse.smarthome.automation.AutomationFactory;
+import org.eclipse.smarthome.automation.Condition;
+import org.eclipse.smarthome.automation.Rule;
+import org.eclipse.smarthome.automation.Trigger;
+import org.eclipse.smarthome.automation.handler.parser.Parser;
+import org.eclipse.smarthome.automation.handler.parser.Status;
+import org.eclipse.smarthome.automation.template.RuleTemplate;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class serves for loading JSON files and parse it to the Rule objects.
+ *
+ * @author Ana Dimova - Initial Contribution
+ *
+ */
+public class RuleJSONParser implements Parser {
+
+    private BundleContext bc;
+    private TemplateManager templateManager;
+    private ModuleTypeManager moduleManager;
+    private AutomationFactory automationFactory;
+    private Logger log;
+
+    public RuleJSONParser(BundleContext bc, TemplateManager templateManager, ModuleTypeManager moduleManager,
```

Comment:
new method  will be added in AutomationFactory:
public Rule createRule(String template_UID, Map<String, ?> configurations);
and validation will be done in this method.